src/components/AddItems.js

```diff
@@ -25,20 +24,34 @@ const AddItems = () => {
     });
   };
 
+  // function to compare new entry with existing entries
+  const compareItems = (currentItem) => {
+    const matches = listContext.userList.filter(
+      (item) =>
+        item.itemName.replace(/[^A-Z0-9]+/gi, '').toLowerCase() ==
+        currentItem.replace(/[^A-Z0-9]+/gi, '').toLowerCase(),
+    );
+    return matches.length < 1;
+  };
+
   // submits state to database
   const handleSubmit = (event) => {
     event.preventDefault();
 
-    itemsRef
-      .add(formData)
-      .then(function () {
-        console.log('submitted!');
-        alert('submitted');
-      })
-      // catches & logs any errors
-      .catch(function (error) {
-        console.error('error adding item to the database!', error);
-      });
+    if (compareItems(formData.itemName)) {
+      itemsRef
+        .add(formData)
+        .then(function () {
+          setFormData({ itemName: '' });
+          alert('submitted');
+        })
+        // catches & logs any errors
+        .catch(function (error) {
+          console.error('error adding item to the database!', error);
+        });
+    } else {
+      alert('this item already exists in the database!');
```

Comment:
I would use a <span> with an error next to the input instead of an alert for this scenario. So the user knows what happened even after clicking on the alert.